android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/DemosFragment.java

```diff
@@ -0,0 +1,112 @@
+package org.physical_web.physicalweb;
+
+import android.app.ListFragment;
+import android.content.ContentResolver;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.Toast;
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+
+/**
+ * This class shows a list of demos.
+ */
+public class DemosFragment extends ListFragment {
+  private static final String TAG = DemosFragment.class.getSimpleName();
+  private HashMap<String, Demo> mDemos;
+  private DemosAdapter mAdapter;
+
+  @Override
+  public View onCreateView(LayoutInflater layoutInflater, ViewGroup container,
+      Bundle savedInstanceState) {
+    getActivity().getActionBar().setDisplayHomeAsUpEnabled(true);
+    View rootView = layoutInflater.inflate(R.layout.fragment_demos, container, false);
+    mAdapter = new DemosAdapter();
+    setListAdapter(mAdapter);
+    mDemos = new HashMap<>(1);
```

Comment:
I think it'd make sense for each to have their own file (some may be bigger than this)  make a new package org...demos (i.e. a new directory) and put them all in there.